### DISCUSSION TITLE
Page change, but on the same route

### DISCUSSION THREAD

**Discussion by Angelisium** [solidjs/solid-router#240] [Q&A] [answered]
Hi 👋

Let's imagine a fictitious simplified routing such as :
https://codesandbox.io/p/sandbox/test-routing-solidjs-9og6zk

When I click on the link, it changes the url but it doesn't reload the content of the page since (for the router) we are still on the same page 😬

How can I fix this?

**Comment by Brendan-csel** [comment, marked as answer]
The `Chapter` function only runs once.  So you can't destructure the params at the top.

Instead do this...
```js
function Chapter(props) {
	const params = useParams();
	return (
		<>
			<p>{params.slug + " : " + params.page}</p>
---

```

**Comment by Angelisium** [comment]
thank you for your answer 💖

edit : in fact, I oversimplified the problem was with `data={DataLoader}` which is called only once...
But it's logic since it does not reload the whole page (as it would be the case if I go from the `/book/:slug` page to the `/book/:slug/:page` page)

But maybe I'm doing something wrong and breaking the reactivity?

In any case, I moved the logic in my component and I removed `useRouteData` and it works 😋
